dex/elrond_dex_farm/src/rewards.rs

```diff
@@ -95,8 +103,12 @@ pub trait RewardsModule: config::ConfigModule + token_supply::TokenSupplyModule
         current_reward_per_share: &Self::BigUint,
         initial_reward_per_share: &Self::BigUint,
     ) -> Self::BigUint {
-        let reward_per_share_diff = current_reward_per_share - initial_reward_per_share;
-        amount * &reward_per_share_diff / self.division_safety_constant().get()
+        if current_reward_per_share > initial_reward_per_share {
```

Comment:
when merging, initial_reward_per_share is rounded to upper value, that's why we need the check

Reply:
wouldn't it be safer to round to initial_reward_per_share to the lower value ?

Reply:
no, because then the current_reward_per_share - initial_reward_per_share will be bigger than expected and the sc might try to send more rewards than sc have